appkit/src/test/scala/org/ergoplatform/appkit/DHTProverSpec.scala

```diff
@@ -0,0 +1,53 @@
+package org.ergoplatform.appkit
+
+import org.ergoplatform.{ErgoAddressEncoder, Pay2SAddress}
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import scorex.util.encode.Base16
+import sigmastate.eval._
+import sigmastate.interpreter.CryptoConstants
+import sigmastate.serialization.ErgoTreeSerializer
+import special.sigma.GroupElement
+
+class DHTProverSpec extends PropSpec with Matchers
+  with ScalaCheckDrivenPropertyChecks
+  with AppkitTesting
+  with HttpClientTesting {
+
+  property("DHTProver") {
+    val ergoClient = createMockedErgoClient(MockData(Nil, Nil))
+    val g: GroupElement = CryptoConstants.dlogGroup.generator
+    val x = BigInt("187235612876647164378132684712638457631278").bigInteger
+    val y = BigInt("340956873409567839086738967389673896738906").bigInteger
+    val gX:GroupElement = g.exp(x)
+    val gY:GroupElement = g.exp(y)
+    val gXY:GroupElement = gY.exp(x)
```

Comment:
@scalahub shouldn't it be `gX.exp(y)`?

Reply:
It can be either `gX.exp(y)` or `gY.exp(x)` as the operation is commutative, so should not matter.

Reply:
Well, strictly speaking it matters, because it triggered my attention, high enough to write the question. Either fix the order or add a comment.

Reply:
Fixed, also removed ScalaDoc from impl class and put in abstract class.